cob_monitoring/src/cpu_monitor.py

```diff
@@ -261,82 +125,37 @@ def check_core_temps(self):
-                pass
-
-            if retcode != 0:
-                diag_level = DiagnosticStatus.ERROR
-                diag_msgs = [ 'Clock Speed Error' ]
-                diag_vals = [ KeyValue(key = 'Clock Speed Error', value = stderr),
-                              KeyValue(key = 'Output', value = stdout) ]
-                return (diag_vals, diag_msgs, diag_level)
-
-            for index, ln in enumerate(stdout.split('\n')):
-                words = ln.split(':')
-                if len(words) < 2:
-                    continue
-
-                speed = words[1].strip().split('.')[0] # Conversion to float doesn't work with decimal
-                diag_vals.append(KeyValue(key = 'Core %d MHz' % index, value = speed))
-                try:
-                    _ = float(speed)
-                except ValueError:
-                    diag_level = max(diag_level, DiagnosticStatus.ERROR)
-                    diag_msgs = [ 'Clock speed not numeric' ]
+            netdata_cpu_freq = query_netdata('cpu.cpufreq', interval)
+            del netdata_cpu_freq["time"]
 
-            # get max freq
-            p = subprocess.Popen('lscpu | grep "max MHz"',
-                                 stdout = subprocess.PIPE,
-                                 stderr = subprocess.PIPE, shell = True)
-            stdout, stderr = p.communicate()
-            retcode = p.returncode
-            try:
-                stdout = stdout.decode()  #python3
-            except (UnicodeDecodeError, AttributeError):
-                pass
-
-            if retcode != 0:
+            if not netdata_cpu_freq:
                 diag_level = DiagnosticStatus.ERROR
                 diag_msgs = [ 'Clock Speed Error' ]
-                diag_vals = [ KeyValue(key = 'Clock Speed Error', value = stderr),
-                              KeyValue(key = 'Output', value = stdout) ]
+                diag_vals = [ KeyValue(key = 'Clock Speed Error', value = 'Could not fetch data from netdata'),
+                              KeyValue(key = 'Output', value = str(netdata_cpu_freq)) ]
                 return (diag_vals, diag_msgs, diag_level)
 
-            diag_vals.append(KeyValue(key = stdout.split(':')[0].strip(), value = str(stdout.split(':')[1].strip())))
+            for cpu_name, values in netdata_cpu_freq.items():
+                diag_vals.append(KeyValue(key = 'Core %d (MHz)' % int(cpu_name[-1]), value = str(np.mean(values))))
 
-            # get min freq
-            p = subprocess.Popen('lscpu | grep "min MHz"',
-                                 stdout = subprocess.PIPE,
-                                 stderr = subprocess.PIPE, shell = True)
-            stdout, stderr = p.communicate()
-            retcode = p.returncode
-            try:
-                stdout = stdout.decode()  #python3
-            except (UnicodeDecodeError, AttributeError):
-                pass
+            # get max freq
+            netdata_max_cpu_freq = query_netdata_info()
```

Comment:
is this correct?
dont we need to extract the max_cpu_freq from the info - as is done for the num_cores?

Reply:
The varibale name is a bit misleading. Should be of course also `netdata_info`